### PR TITLE
Bump Go and docker/docker

### PR DESCRIPTION
**Pull request by daniponsg**
docker/docker v24 is affected by [CVE-2024-41110 ](https://avd.aquasec.com/nvd/2024/cve-2024-41110)

### CODE REVIEW

File: puller.go [hunk 1]
```diff
@@ -31,7 +32,7 @@ type RepositoryConfig struct {
 
 	// Auth provides the default docker library's field to authenticate and will be used for pulling.
 	// Usually Username & Password fields should be filled.
-	Auth types.AuthConfig
+	Auth registry.AuthConfig
```
Comment (scheca) [inline puller.go:35]: Hey! nice to see activity in this project 🔝 
Would this be a breaking change? It's an exported field of an exported struct and its type has changed from `types.AuthConfig` to `registry.AuthConfig`

Comment (daniponsg) [reply puller.go:35]: Hey @scheca! Glad to see you here :)

Yes, it could be a breaking change. Bumping is being done due to vulnerabilities.

Comment (yaroslavcev) [reply puller.go:35]: Good point to note, thanks @scheca  The `types.AuthConfig` type was defined as `type AuthConfig = registry.AuthConfig` in the previous version, so it doesn't look critical

**Review by yaroslavcev — APPROVED** [review]
LGTM

**Pull request merged**